pkg/instrumentation/defaultinstrumentation.go

```diff
@@ -77,9 +77,8 @@ func getDefaultInstrumentation(agentConfig *adapters.CwaConfig, isWindowsPod boo
 
 	// set protocol by checking cloudwatch agent config for tls setting
 	exporterPrefix := http
-	if agentConfig != nil {
-		appSignalsConfig := agentConfig.GetApplicationSignalsConfig()
-		if appSignalsConfig != nil && appSignalsConfig.TLS != nil {
+	if isApplicationSignalsEnabled(agentConfig) {
```

Comment:
nit: Instead of evaluating this function each time, consider storing it in a variable and just passing that around.
```suggestion
    isApplicationSignalsEnabled := agentConfig != nil && agentConfig.GetApplicationSignalsConfig() != nil
	if isApplicationSignalsEnabled {
```

Reply:
updated